src/bika/lims/validators.py

```diff
@@ -642,7 +642,8 @@ class ResultOptionsValueValidator(object):
     def __call__(self, value, *args, **kwargs):
         # Result Value must be floatable
         if not api.is_floatable(value):
-            return _t(_("Result Value must be a number"))
+            if not api.is_uid(value):
+                return _t(_("Result Value must be a number or valid UID"))
```

Comment:
Confusing, because when I type in a number with 37 digits it will fail as well

Reply:
To resolve this the underlying field could enforce a max field length required.